dlc-manager/src/manager.rs

```diff
@@ -592,6 +625,78 @@ where
         Ok(())
     }
 
+    /// Manually close a contract with the oracle attestations.
+    pub fn close_confirmed_contract(
+        &mut self,
+        contract_id: &ContractId,
+        attestations: Vec<(usize, OracleAttestation)>,
+    ) -> Result<Contract, Error> {
+        let contract = get_contract_in_state!(self, &contract_id, Confirmed, None::<PublicKey>)?;
+        let contract_infos = &contract.accepted_contract.offered_contract.contract_info;
+        let adaptor_infos = &contract.accepted_contract.adaptor_infos;
+
+        // find the contract info that matches the attestations
+        if let Some((contract_info, adaptor_info)) =
+            contract_infos.iter().zip(adaptor_infos).find(|(c, _)| {
+                let matches = c
+                    .oracle_announcements
+                    .iter()
+                    .filter(|o| {
+                        attestations
+                            .iter()
+                            .any(|(_, a)| o.oracle_event.oracle_nonces == a.nonces())
```

Comment:
In order for this to work, you also need to make sure that the oracle index provided as the first parameter of the tuples within `attestations` are actually correct, so you should not be looking up but instead taking the provided index. I still think it should be either the user passes the index of the contract info to use, or passes no index at all and the oracle index is also found within the library. I think the former might be better because the user would probably have the information if they managed to get the attestations in the first place.

Reply:
I did the lookup by index, isn't very hard for the library to do imo

Reply:
@benthecarman so you want to keep it like it is now?

Reply:
@benthecarman I'm really just waiting for you to confirm whether you want to keep it as is or change it before I approve and merge :)

Reply:
Oh sorry missed these, yeah I think this is fine as is. Just makes it easier for the end user